segmentstore/storage/impl/src/main/java/io/pravega/segmentstore/storage/impl/bookkeeper/BookKeeperServiceRunner.java

```diff
@@ -253,10 +258,37 @@ private void initializeZookeeper() throws Exception {
     private void runBookies() throws Exception {
         log.info("Starting Bookie(s) ...");
         for (int bkPort : this.bookiePorts) {
-            this.servers.add(runBookie(bkPort));
+            this.servers.add(runBookieWithRetry(bkPort));
         }
     }
 
+    private BookieServerAndResources runBookieWithRetry(int bkPort) throws Exception {
+        int count = 0;
+        boolean retry = true;
+        BookieServerAndResources bookieServerAndResources = null;
+        while (count < retries && retry) {
+            try {
+                log.info("Starting bookie with retry value of :" + count);
+                bookieServerAndResources = runBookie(bkPort);
+                retry = false;
+            } catch (BindException e) {
+                if ( ++count >= retries ) {
+                    e.printStackTrace();
+                    throw new BookieException(1) {
```

Comment:
What is the point of subclassing this exception and overriding a single method only to have it do exactly what the super method does?

Reply:
Will refactor and remove this.